specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json

```diff
@@ -50,35 +50,47 @@
         "modelAsString": true
       }
     },
-    "ManagedIntegrationRuntime": {
-      "x-ms-discriminator-value": "Managed",
-      "description": "Managed integration runtime, including managed elastic and managed dedicated integration runtimes.",
+    "ManagedIntegrationRuntimeBase": {
+      "description": "Managed integration runtime base.",
       "type": "object",
       "allOf": [
         {
           "$ref": "#/definitions/IntegrationRuntime"
         }
       ],
       "properties": {
-        "state": {
-          "description": "Integration runtime state, only valid for managed dedicated integration runtime.",
-          "$ref": "#/definitions/IntegrationRuntimeState",
-          "readOnly": true
-        },
         "typeProperties": {
           "description": "Managed integration runtime properties.",
           "x-ms-client-flatten": true,
           "$ref": "#/definitions/ManagedIntegrationRuntimeTypeProperties"
         },
         "managedVirtualNetwork": {
-          "description": "Managed Virtual Network reference.",
+          "description": "Managed integration runtime managed virtual network.",
+          "x-ms-client-flatten": true,
           "$ref": "../datafactory.json#/definitions/ManagedVirtualNetworkReference"
         }
       },
       "required": [
         "typeProperties"
       ]
     },
+    "ManagedIntegrationRuntime": {
```

Comment:
I would suggest to inherit `IntegrationRuntime` directly rather than two layers inheritance.

Reply:
The reason we add two layer is to synchronize common properties between Data Factory and Synapse.